diff --git a/nixos/modules/system/boot/loader/limine/limine-install.py b/nixos/modules/system/boot/loader/limine/limine-install.py
index d7dd5764ef26..1d7169878b92 100644
--- a/nixos/modules/system/boot/loader/limine/limine-install.py
+++ b/nixos/modules/system/boot/loader/limine/limine-install.py
@@ -148,12 +148,13 @@ def config_entry(levels: int, bootspec: BootSpec, label: str, time: str) -> str:
         entry += f'module_path: ' + get_kernel_uri(bootspec.initrd) + '\n'
 
     if bootspec.initrdSecrets:
-        initrd_secrets_path = limine_dir + '/kernels/' + os.path.basename(toplevel) + '-secrets'
-        os.makedirs(initrd_secrets_path)
+        base_path = str(limine_dir) + '/kernels/'
+        initrd_secrets_path = base_path + os.path.basename(bootspec.toplevel) + '-secrets'
+        if not os.path.exists(base_path):
+            os.makedirs(base_path)
 
-        old_umask = os.umask()
-        os.umask(0o137)
-        initrd_secrets_path_temp = tempfile.mktemp(os.path.basename(toplevel) + '-secrets')
+        old_umask = os.umask(0o137)
+        initrd_secrets_path_temp = tempfile.mktemp(os.path.basename(bootspec.toplevel) + '-secrets')
 
         if os.system(bootspec.initrdSecrets + " " + initrd_secrets_path_temp) != 0:
             print(f'warning: failed to create initrd secrets for "{label}"', file=sys.stderr)
